frontend/src/components/common/Resource/ScaleButton.tsx

```diff
@@ -0,0 +1,222 @@
+import DialogActions from '@material-ui/core/DialogActions';
+import DialogContent from '@material-ui/core/DialogContent';
+import DialogTitle from '@material-ui/core/DialogTitle';
+import Fab from '@material-ui/core/Fab';
+import Grid from '@material-ui/core/Grid';
+import IconButton from '@material-ui/core/IconButton';
+import MuiInput from '@material-ui/core/Input';
+import { makeStyles, styled, useTheme } from '@material-ui/core/styles';
+import Tooltip from '@material-ui/core/Tooltip';
+import React from 'react';
+import { useTranslation } from 'react-i18next';
+import { useDispatch } from 'react-redux';
+import { useLocation } from 'react-router-dom';
+import { KubeObject } from '../../../lib/k8s/cluster';
+import { CallbackActionOptions, clusterAction } from '../../../redux/actions/actions';
+import { LightTooltip } from '../Tooltip';
+
+interface ScaleButtonProps {
+  item: KubeObject;
+  options?: CallbackActionOptions;
+}
+
+export default function ScaleButton(props: ScaleButtonProps) {
+  const dispatch = useDispatch();
+  const { item, options = {} } = props;
+  const [openDialog, setOpenDialog] = React.useState(false);
+  const [visible, setVisible] = React.useState(false);
+  const location = useLocation();
+  const { t } = useTranslation('resource');
+
+  async function updateFunc(numReplicas: number) {
+    try {
+      await item.scale(numReplicas);
+    } catch (err) {
+      throw err;
+    }
+  }
+
+  const applyFunc = React.useCallback(updateFunc, [item]);
+
+  function handleSave(numReplicas: number) {
+    const cancelUrl = location.pathname;
+    const itemName = item.metadata.name;
+
+    setOpenDialog(false);
+
+    // setOpenDialog(false);
+    dispatch(
+      clusterAction(() => applyFunc(numReplicas), {
+        startMessage: t('Scaling {{ itemName }}…', { itemName }),
+        cancelledMessage: t('Cancelled scaling {{ itemName }}.', { itemName }),
+        successMessage: t('Scaled {{ itemName }}.', { itemName }),
+        errorMessage: t('Failed to scale {{ itemName }}.', { itemName }),
+        cancelUrl,
+        errorUrl: cancelUrl,
+        ...options,
+      })
+    );
+  }
+
+  function handleClose() {
+    setOpenDialog(false);
+  }
+
+  React.useEffect(() => {
+    if (item) {
+      item
+        .getAuthorization('update')
+        .then((result: any) => {
+          if (result.status.allowed) {
+            setVisible(true);
+          }
+        })
+        .catch((err: Error) => {
+          console.error(`Error while getting authorization for edit button in ${item}:`, err);
+          setVisible(false);
+        });
+    }
+  }, [item]);
+
+  if (!visible || !['Deployment', 'StatefulSet', 'ReplicaSet'].includes(item.kind)) {
+    return null;
+  }
+
+  return (
+    <React.Fragment>
+      <Tooltip title={t('frequent|Scale') as string}>
+        <IconButton aria-label={t('frequent|scale')} onClick={() => setOpenDialog(true)}>
+          <Icon icon="mdi:overscan" />
```

Comment:
Not sure this is a good icon. Looks close to a fullscreen button too me.

Reply:
Changed to content-copy